pdns/dnsdist.cc

```diff
@@ -1475,6 +1479,14 @@ try
       break;
     }
   }
+
+#ifdef HAVE_SYSTEMD
+  char *sock;
+  sock = getenv("NOTIFY_SOCKET");
+  if(sock != NULL) // Are we indeed running inside of systemd?
+    g_cmdLine.beSupervised=true;
+#endif
```

Comment:
not if your actually running inside systemd. There is no other way (as daemonizing will break systemd's detection of a startup and not being supervised will spawn a console with a closed stdin, leading to program termination) to do this. We could just `exit(1)` here and print a message asking the user to start with `--supervised`